src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionService.java

```diff
@@ -91,6 +91,41 @@ public interface SnowflakeConnectionService {
    */
   boolean isTableCompatible(String tableName);
 
+  /**
+   * Check whether the user has the role privilge to do schema evolution and whether the schema
```

Comment:
nit: *privilege
For my education, which role gives schema evolution?

Reply:
I'm not sure if I understand the question.

The role should have one of "EVOLVE SCHEMA", "ALL" or "OWNERSHIP" privilege to do schema evolution.